File: src/SOP8/Utility/Response/Resource/ID.cs
namespace Response.Resource
{
    public class ID
    {
        public enum LanguageTypes { ko, en }

        private static LanguageTypes m_targetLanguage = LanguageTypes.ko;

        public static LanguageTypes TargetLanguage
        {
            get { return m_targetLanguage; }
            set { m_targetLanguage = value; }
        }

        public static MessageMap Get<T>(string strTag) where T : MessageMap, new()
        {
            T t = new T();
            return t.Get(m_targetLanguage).Get(strTag);
        }
    }
}

File: src/SOP8/Egovframework/Base/base-utility/src/main/java/egovframework/base/response/resource/ID.java
package egovframework.base.response.resource;

import egovframework.base.response.resource.MessageMap;

public class ID {

    public enum LanguageTypes {
        ko, en
    }

    private static LanguageTypes targetLanguage = LanguageTypes.ko;

    public static LanguageTypes getTargetLanguage() {
        return targetLanguage;
    }

    public static void setTargetLanguage(LanguageTypes value) {
        targetLanguage = value;
    }

    public static <T extends MessageMap> MessageMap get(Class<T> clazz, String strTag) {
        try {
            T t = clazz.getDeclaredConstructor().newInstance();
            return t.get(targetLanguage).get(strTag);
        } catch (Exception e) {
            throw new RuntimeException("Failed to create instance of " + clazz.getName(), e);
        }
    }
}